akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/scaladsl/ClusterActorSet.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed.scaladsl
+
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Behavior
+import akka.actor.typed.Extension
+import akka.actor.typed.ExtensionId
+import akka.annotation.DoNotInherit
+import akka.cluster.sharding.ShardRegion.EntityId
+import akka.cluster.sharding.typed.internal.ClusterActorSetImpl
+
+import scala.concurrent.duration.FiniteDuration
+import akka.util.JavaDurationConverters._
+import com.typesafe.config.Config
+
+object ClusterActorSetSettings {
+
+  def apply(system: ActorSystem[_]): ClusterActorSetSettings = {
+    apply(system.settings.config.getConfig("akka.cluster.actor-set"))
+  }
+
+  def apply(config: Config): ClusterActorSetSettings = {
+    val numberOfShards = config.getInt("number-of-shards")
+    val keepAliveInterval = config.getDuration("keep-alive-interval").asScala
+
+    new ClusterActorSetSettings(keepAliveInterval, numberOfShards)
+  }
+
+}
+
+class ClusterActorSetSettings private[akka] (val keepAliveInterval: FiniteDuration, val numberOfShards: Int) {
+  def withKeepAliveInterval(keepAliveInterval: FiniteDuration): ClusterActorSetSettings =
```

Comment:
This implementation is :+1: for now.

We've talked about making 'remember entities' nicer in the future. If we succeed in that I could see us use that instead to achieve the 'keep alive' aspect of this feature. Given that, should we make the pinging more of an implementation detail and perhaps mark the interval setting as may-change?

Reply:
Good point

Reply:
Thinking a bit more about this, I'm not entirely sure why we currently need the periodic ping and can't just use ddata remember-entities, the size of the set is expected to be low...

Reply:
The reason why it is needed is to start the set of actors on a full cluster stop. 

I think we could arrive there by replacing the constant pinging with a ping pong instead, once a node has ping-ponged each entity parent it could stop it self instead of periodically pinging.

Reply:
we should start without relying on remember entities, but good point about that the impl may change

Reply:
You need to keep sending the pings because when you lose a node you need the means to restart the workers on the remaining nodes.

Reply:
that is what remember entities handles, but we should anyway not rely on that feature for now